### ISSUE TITLE
BUG-000176156 [Popover] Keyboard is not accessible from a Calcite Input on horizontally oriented mobile device

### ISSUE THREAD

**Issue by annelfitz** [Esri/calcite-design-system#13426]
**monday.com sync:** #10738255808

### Check existing issues

- [x] I have [checked for existing issues](https://github.com/Esri/calcite-design-system/issues) to avoid duplicates

### Which Component

Input, Popover

### Actual Behavior

When a calcite input is used within a popover (towards the bottom) on a mobile device, and the device is rotated horizontally, the keyboard is not accessible from the input. It will auto-close and never fully display.

https://github.com/user-attachments/assets/cf1143a4-28f2-40d9-8822-82865c4cd99a

### Expected Behavior

The keyboard should be accessible from the input even when used within a popover component.

### Reproduction Sample

https://codepen.io/annefitz/pen/YPqBzxa?editors=1010 (open on a mobile device, outside of codepen if possible)

### Reproduction Steps

1. Copy the codepen to a locally hosted version of the app. 
2. Start the macOS simulator and open the application in Safari (or open on a physical iOS device)
3. Orient the mobile device horizontally
4. Try to focus the input, observe that the virtual keyboard auto-closes / never fully displays

### Reproduction Version

latest

### Relevant Info

Only reproducible on iOS devices

Bug was initially reported in the JS SDK here: https://desktopqa/SFTriage/#/BUG-000176156

More information here: https://devtopia.esri.com/WebGIS/arcgis-web-components/issues/6906#issuecomment-5956612

### Priority impact

impact - p2 - want for an upcoming milestone

### Calcite package

- [x] @esri/calcite-components
- [ ] @esri/calcite-components-react
- [ ] @esri/calcite-design-tokens
- [ ] @esri/calcite-ui-icons
- [ ] @esri/eslint-plugin-calcite-components

### Esri team

ArcGIS Maps SDK for JavaScript<div><div><div>Typing in text boxes is not possible when using the Expand widget in &#39;floating&#39; mode on mobile devices with ArcGIS Maps SDK for JavaScript 4.32.</div></div>
<span><b>Salesforce ID:</b> <a href="https://desktopqa/SFTriage?sfid=BUG-000176156">BUG-000176156</a></span><br/>
<span><b>Salesforce Submitter:</b> Sreyoshi Paul</span><br/>
<span><b>Salesforce Submit Date:</b> 4/30/2025 6:18 PM</span><br/>
<span><b>Salesforce Bug Type:</b> Failure/Error</span><br/>
<span><b>Salesforce Severity:</b> Low</span><br/>
<span><b>Steps to Repro:</b> <p><strong>Steps to Reproduce:</strong></p><ol><li>Copy the code from here: https://codepen.io/SreyoshiP/pen/bNNeyjw to an html file and host the sample application in a local server and launch the app on the mobile devices to test this issue.</li><li>Press the expand button on the top right </li><li>Try to type in a text box after selecting a text box <ol><li>iPhone: Notice that keyboard does not appear, and typing is not possible </li><li>Android: Notice that after tapping in the text box, the cursor is immediately dislodged and cannot be typed.</li><li>iPad: Notice that after tapping in the text box, the cursor is immediately dislodged and cannot be typed. </li></ol></li></ol><p><br></p><p><strong>Other Information:</strong></p><ol><li>This is a simple application with no geospatial data inside </li><li>In iOS devices, the issue is only reproducible in hosted application. </li><li>If opening directly from Codepen, the Expand widget works as expected and no issue is observed.</li><li>Expand widget is set to floating mode </li><li>A form with multiple text box and select box is placed inside the Expand widget using the components &lt;arcgis-placement&gt; and &lt;form&gt; </li><li>The customer reported this issue with iPhone, iPad and Android </li><li>Customer observed this issue using Editor widget (with a long list of editable fields) inside the Expand widget.  </li><li>However, I tested this further and observed that the issue is reproducible with any form with text box where input values can be typed. Also, I tested just the Editor widget without Expand component and the issue did not reproduce. </li><li>Hence, the issue is specific to the Expand widget when any form with text box is present inside it in “floating” mode</li></ol><p><br></p><p><strong>Customer Data Specific Bug:</strong></p><p>No</p><p><br></p><p><strong>Repro Environment Details:</strong></p><p>Android version 14.0.0.933 (EX01V60P01), iPhone 16 Pro iOS version 18.3.2 (22D82), iPad Mini (6th generation) iPad OS version 18.4.1 (Chrome &amp; Safari browser)</p><p><br></p><p><strong>Sample Environment Access Information:</strong></p><p>Sample application: https://codepen.io/SreyoshiP/pen/bNNeyjw</p><p><br></p><p><strong>Development/PSIRT request?</strong></p><p>No</p><p><br></p><p><strong>Attached Files: </strong><span style="background-color: rgb(255, 255, 255); color: rgb(68, 68, 68);">Yes</span></p><p><strong>Attached Video: </strong>No</p><p><strong>Attached Logs: </strong>No</p><p><strong>Attached Data: </strong>Yes</p><p><strong>Reproducible in Latest Version: </strong>Yes</p></span><br/>
<span><b>Repro Data:</b> \\esri.com\sf_filestore\PRD\Attachments\Defects\BUG-000176156</span><br/>
<span><b>Work Around:</b> (n/a)</span><br/>
<span><b>Product:</b> Calcite Design System</span><br/>
<span><b>Functional Category:</b> Calcite Design System</span><br/>
<span><b>Client Platform:</b> (n/a)</span><br/>
<span><b>Version Found:</b> 4.32</span><br/>
<span><b>Planned Version Fixed:</b> (n/a)</span><br/>
<span><b>Comment:</b> (n/a)</span><br/>
</div>

**Comment by geospatialem** [comment]
Spike to conduct additional testing with `5.0` to confirm behavior with recent `popover` and focus modifications/changes.

**Comment by isaacbraun** [comment]
Spike completed. Tested using a local HTML file served over the network to my iPhone in Safari and Chrome using `5.0.0-next.74`, and observed the behavior is still as described. I was also unable to focus the input in portrait mode.

**Comment by driskull** [comment]
Whats happening here is that the virtual keyboard shifts the page by scrolling it. When that happens the form element is hidden and then the popover is closed which then closes the virtual keyboard because it is no longer visible or focused.

The reason it happens is because we use the floating-ui 'hide' modifier: https://floating-ui.com/docs/hide.

This modifier hides the popover when its reference-element is no longer in view. 

We could:

- Add an option on popover or all floating elements to disable the hide modifier.
- Add an option on popover or all floating elements to configure the hide modifier's strategy from `referenceHidden` to `escaped`. Something similar to what we have with `overlayPositioning`. The options could be "referenceHidden, escaped, none"? Something like `hideOn: "reference-hidden" | "escape" | "none" = "reference-hidden";`
- Stop using the hide modifier?
- Magic 🪄

**Comment by driskull** [comment]
1. referenceHidden (Anchor is out of view)What it means: The reference (or trigger) element has scrolled out of view or is fully clipped by its container. The floating element effectively appears to be attached to nothing.Use Case: When a user scrolls a page and the button that opened the dropdown disappears from the screen, you usually want the dropdown to disappear too.
2. escaped (Floating element is out of view)What it means: The floating element itself has been pushed outside of the clipping boundary (e.g., the edge of the screen or a scroll container).Use Case: When the floating element gets clipped or awkwardly cut off at the edge of the viewport, and you prefer to hide it completely rather than show a partially visible dropdown.

**Comment by isaacbraun** [comment]
Verified on `5.2.0-next.24` via https://codepen.io/isaacbraun/pen/jEyXNoL that the nested Input can be focused and remains within view on mobile devices. Tested on:
- Portrait and horizontal view modes
- Chrome and Safari
